### DISCUSSION TITLE
OMR kernel on router and VPS don’t mat after update

### DISCUSSION THREAD

**Discussion by Melania95** [Ysurac/openmptcprouter#3295] [General]
Hi friends, I finally managed to install the new version with 0.60 with kernel 6.1. It works well and I have aggregation but the message "MPTCP kernel on router and VPS don't match..." appears on the VPS (i have also tryed 0.60 5.4 legacy and it's all ok but OMR-Bypass not work well there). I have a VPS Serverplan based on Ubuntu 20.04 and I tried to issue this command trying to update it: wget -O - https://www.openmptcprouter.com/server/ubuntu20.04-x86_64.sh | KERNEL="6.1" sh but I haven't solved it...It updates everything even if it says "can'f find unit file for serviceiperf3". Could it be that the Ubuntu VPS is no longer supported but only with Debian? Or is there some particular setting to do? How could I solve it? Thank you all...
P.S. In any case everything also works well and aggregates and is even faster than before and finally with this kernel the OMR-Bypass also seems to work...

**Comment by Ysurac** [comment]
What is the result of ```uname -a``` on the VPS ?

**Reply by Melania95** [reply]
Hello master and thank you so much for everything you do...I entered ssh in the Vps and typed that command you asked for and i have: Linux v023kml-rssbgr 5.4.207-mptcp #1 SMP Sun Jul 24 14:39:44 UTC 2022 x86_64 x8                                                                                                                                                                                                                                                                                                                                                           6_64 x86_64 GNU/Linux

**Reply by Ysurac** [reply]
Aggregation can't work with this kernel, a kernel 6.1 (or more) is needed on VPS side.
You can try something like that: https://www.linuxcapable.com/how-to-install-latest-linux-kernel-on-ubuntu-linux/
Ubuntu is not supported for now, too much work... If this guide work, I will add the changes in the script.

**Reply by Melania95** [reply]
Ok thank you...So now it's aggregating but using kernel 5.4 right? Because the speed is there anyway and from "bandwith in real time" I see that in any case by doing a speed test every connection "aggregates" and works fast...I would like to leave this setting as it is because if I install the version with kernel 5.4 OMR bypass does not work well. Can I leave it like this if it actually works with the 5.4 kernel? Thank you...

**Reply by Ysurac** [reply]
Hum, yes I forgot that I force kernel 5.4 to use MPTCPv1 when a kernel 5.15 or more is used on router. So yes it may work, but not really tested much. If it's working, keep it.

**Reply by Melania95** [reply]
Thank you so much for all....
Martina.

**Reply by Melania95** [reply]
I will try to install kernel 6.7 on Ubuntu and if successful I will update you...

**Comment by talitha77** [comment]
I have the same error.. updated to v6.0 on the router and kernel 6.1 debian vps.. but getting Direct output (145.224.101.31)
MPTCP kernel on router and VPS doesn't match
Version 0.1028 5.4.207-mptcp
(Latest available version 0.1030)

**Reply by Ysurac** [reply]
v6.0 on the router ? there is a v0.60 that is available with 5.4 or 6.1 kernel.
On the VPS check with ```uname -a``` what kernel you have.

**Reply by talitha77** [reply]
Linux vps-908a98c7 6.1.0-22-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.94-1 (2024-06-21) x86_64 GNU/Linux

**Reply by talitha77** [reply]
I've loaded v.060 with 5.4 kernel this end and downgraded the vps thus : wget -O - https://www.openmptcprouter.com/server/debian-x86_64.sh | KERNEL="5.4" sh
and nothing works at all now..  i backed up config and restored.. i should have left it alone.. it was fine this morning.. i just thought i'd update the rpi from v.059 for fun

**Reply by Ysurac** [reply]
You should open your own ticket and put a screenshot of your status page.
And check kernel on VPS again if you change (it's not a good idea...)

**Reply by talitha77** [reply]
fortunately i had created a dmg of the sd card last week, so I restored and am back in business thanks.  I'm back on v.059 but its all working so I'll walk away :)
